### ISSUE TITLE
Magic number `100%N` used as "fuel" in reflexive normalization (?)

### ISSUE THREAD

**Issue by pi8027** [math-comp/algebra-tactics#57]
I should investigate this at some point, but I'm not sure if 100 is large enough:
https://github.com/math-comp/algebra-tactics/blob/0b8356c0168fc91b8355d2507343f3d5fdacff05/theories/ring.v#L654
Coq uses 1000 instead:
https://github.com/coq/coq/blob/110921a449fcb830ec2a1cd07e3acc32319feae6/theories/setoid_ring/Ring_tac.v#L340

I think we should at least give a constant name to this number as in Coq.

**Comment by thery** [comment]
Oh my god no :scream:   Could you use some binary encoding so to give as fuel something really big.

**Comment by pi8027** [comment]
The actual code that requires this fuel is, I guess, these functions in `setoid_ring`: https://github.com/coq/coq/blob/110921a449fcb830ec2a1cd07e3acc32319feae6/theories/setoid_ring/Ring_polynom.v#L467-L499

I'm not going to change them. But, it seems to me that [`(10*10*10)%nat`](https://github.com/coq/coq/blob/110921a449fcb830ec2a1cd07e3acc32319feae6/theories/setoid_ring/Ring_tac.v#L340) is an effective technique to reduce the term size, including those obtained during reduction.

**Comment by thery** [comment]
is this the code that handles the substitution of `H1` in  `ring [H1]`?

**Comment by pi8027** [comment]
> is this the code that handles the substitution of `H1` in `ring [H1]`?

Yes, I think so, and my main problem is that I have no idea how big this fuel should be.

> Can you try `x^101 = 1` with `H1 : x = 1`?

It works without any problem:
```coq
Goal forall (R : comRingType) (x : R), x = 1 -> x ^+ 101 = 1.
Proof. by move=> R x; ring. Qed.

Goal forall (R : comRingType) (x : R), x = 1 -> x ^+ 10000 = 1.
Proof. by move=> R x; ring. Qed.
```

**Comment by pi8027** [comment]
> is this the code that handles the substitution of `H1` in `ring [H1]`?

Yes, I think so, and my main problem is that I have no idea how big this fuel should be.

> Can you try `x^101 = 1` with `H1 : x = 1`?

It works without any problem:
```coq
Goal forall (R : comRingType) (x : R), x = 1 -> x ^+ 101 = 1.
Proof. by move=> R x; ring. Qed.

Goal forall (R : comRingType) (x : R), x = 1 -> x ^+ 10000 = 1.
Proof. by move=> R x; ring. Qed.
```

**Comment by thery** [comment]
you don't need to say explicitely to use the assumption?

**Comment by thery** [comment]
this is dangerous!  `x = x^2 -> x  = x` `ring` will use all the fuel.

**Comment by pi8027** [comment]
Aha, I thought it was more intelligent. I will try to produce such an example.
